### PR TITLE
Speed up bfloat16 to float conversion

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli and @havardpe PR

### CODE REVIEW

File: searchlib/src/vespa/searchlib/tensor/temporary_vector_store.cpp [hunk 1]
```diff
@@ -1,30 +1,43 @@
...
 convert_cells(ArrayRef<ToType> space, TypedCells cells) noexcept
 {
-    assert(cells.size == space.size());
-    auto old_cells = cells.typify<FromType>();
+    auto old_cells = cells.unsafe_typify<FromType>();
     ToType *p = space.data();
     for (FromType value : old_cells) {
-        ToType conv(value);
-        *p++ = conv;
+        *p++ = value;
```
Comment (vekterli) [inline searchlib/src/vespa/searchlib/tensor/temporary_vector_store.cpp:27]: Are all other cell types well-defined with just a cast? Also, should we have a `static_cast` here to make the conversion more obvious?

File: searchlib/src/vespa/searchlib/tensor/temporary_vector_store.cpp [hunk 2]
```diff
@@ -1,30 +1,43 @@
...
     }
     return space;
 }
 
+template<>
+ConstArrayRef<float>
+convert_cells<vespalib::BFloat16, float>(ArrayRef<float> space, TypedCells cells) noexcept
+{
+    static const IAccelrated & accelrator = IAccelrated::getAccelerator();
+    accelrator.convert_bfloat16_to_float(reinterpret_cast<const uint16_t *>(cells.data), space.data(), space.size());
```
Comment (vekterli) [inline searchlib/src/vespa/searchlib/tensor/temporary_vector_store.cpp:37]: Nit: `s/accelrator/accelerator` 👀

**Review by vekterli — DISMISSED** [review]
:+1: cross-checked with Godbolt, and the bf16 conversion routine gets auto-vectorized as expected

**Review by havardpe — DISMISSED** [review]
I think this violates some aliasing rules, but hopefully that does not matter

File: searchlib/src/vespa/searchlib/tensor/temporary_vector_store.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/tensor/temporary_vector_store.cpp]: The type are double, float, BFloat16 and Int8Float. As far as I can tell they should be fine.

File: searchlib/src/vespa/searchlib/tensor/temporary_vector_store.cpp (hunk 2 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/tensor/temporary_vector_store.cpp]: Fixed

**Pull request merged**